Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -268,31 +244,451 @@ def embProdEmbOfIsAlgebraic [Algebra E K] [IsScalarTower F E K] (halg : Algebra.
+  · exact separable_of_mem_isSeparable F E <| mem_adjoin_simple_self F x
+  · have h := hsep.isIntegral
+    haveI := adjoin.finiteDimensional h
+    rwa [← finSepDegree_eq_finrank_iff,
+      finSepDegree_adjoin_simple_eq_finrank_iff F E x h.isAlgebraic]
+
+/-- If `E / F` and `K / E` are both separable extensions, then `K / F` is also separable. -/
+theorem IsSeparable.trans [Algebra E K] [IsScalarTower F E K]
+    [IsSeparable F E] [IsSeparable E K] : IsSeparable F K := (isSeparable_def F K).2 fun x ↦ by
+  let f := minpoly E x
+  let E' : IntermediateField F E := adjoin F f.frange
+  haveI : FiniteDimensional F E' := finiteDimensional_adjoin fun x _ ↦ IsSeparable.isIntegral F x
+  let g : E'[X] := f.toSubring E'.toSubring (subset_adjoin F _)
+  have h : g.map (algebraMap E' E) = f := f.map_toSubring E'.toSubring (subset_adjoin F _)
+  clear_value g
+  have hx : x ∈ restrictScalars F E'⟮x⟯ := mem_adjoin_simple_self _ x
+  have hzero : aeval x g = 0 := by
+    simpa only [← h, aeval_map_algebraMap] using minpoly.aeval E x
+  have halg : IsIntegral E' x := (IsSeparable.isAlgebraic F E).trans
+    (IsSeparable.isAlgebraic E K) x |>.isIntegral.tower_top
+  have hsep : f.Separable := IsSeparable.separable E x
+  rw [← h, separable_map] at hsep
+  replace hsep := hsep.of_dvd <| minpoly.dvd _ _ hzero
+  haveI : IsSeparable F E' := isSeparable_tower_bot_of_isSeparable F E' E
+  haveI := (isSeparable_adjoin_simple_iff_separable _ _).2 hsep
+  haveI := adjoin.finiteDimensional halg
+  haveI : FiniteDimensional F E'⟮x⟯ := FiniteDimensional.trans F E' E'⟮x⟯
+  have := finSepDegree_mul_finSepDegree_of_isAlgebraic F E' E'⟮x⟯ (IsSeparable.isAlgebraic _ _)
+  rw [finSepDegree_eq_finrank_of_isSeparable F E',
+    finSepDegree_eq_finrank_of_isSeparable E' E'⟮x⟯,
+    FiniteDimensional.finrank_mul_finrank F E' E'⟮x⟯,
+    eq_comm, finSepDegree_eq_finrank_iff F E'⟮x⟯] at this
+  change IsSeparable F (restrictScalars F E'⟮x⟯) at this
+  exact separable_of_mem_isSeparable F K hx
+
+/-- If `x` and `y` are both separable elements, then `F⟮x, y⟯ / F` is a separable extension.
+As a consequence, any rational function of `x` and `y` is also a separable element. -/
+theorem IntermediateField.isSeparable_adjoin_pair_of_separable {x y : E}
+    (hx : (minpoly F x).Separable) (hy : (minpoly F y).Separable) :
+    IsSeparable F F⟮x, y⟯ := by
+  rw [← adjoin_simple_adjoin_simple]
+  replace hy := hy.map_minpoly F⟮x⟯
+  rw [← isSeparable_adjoin_simple_iff_separable] at hx hy
+  exact IsSeparable.trans F F⟮x⟯ F⟮x⟯⟮y⟯
+
+namespace Field
+
+/-- If `x` and `y` are both separable elements, then `x * y` is also a separable element. -/
+theorem separable_mul {x y : E} (hx : (minpoly F x).Separable) (hy : (minpoly F y).Separable) :
+    (minpoly F (x * y)).Separable :=
+  haveI := isSeparable_adjoin_pair_of_separable F E hx hy
+  separable_of_mem_isSeparable F E <| F⟮x, y⟯.mul_mem (subset_adjoin F _ (Set.mem_insert x {y}))
+    (subset_adjoin F _ (Set.mem_insert_of_mem x rfl))
```

Comment:
(untested) some people might prefer API lemmas over defeq, but I personally think there's no harm exploiting defeq here (same in `separable_add` below) ...
```suggestion
  separable_of_mem_isSeparable F E <| F⟮x, y⟯.mul_mem (subset_adjoin F _ (.inl rfl))
    (subset_adjoin F _ (.inr rfl))
```